src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CompoundAssignmentOperator.cs

```diff
@@ -294,13 +294,38 @@ private BoundIndexerAccess TransformIndexerAccess(BoundIndexerAccess indexerAcce
             // tempy = Y()
             // tempc[tempx, tempy, 123] = tempc[tempx, tempy, 123] + 1;
 
-            ImmutableArray<BoundExpression> rewrittenArguments = VisitList(indexerAccess.Arguments);
-
             SyntaxNode syntax = indexerAccess.Syntax;
             PropertySymbol indexer = indexerAccess.Indexer;
+            ImmutableArray<int> argsToParamsOpt = indexerAccess.ArgsToParamsOpt;
             ImmutableArray<RefKind> argumentRefKinds = indexerAccess.ArgumentRefKindsOpt;
+
+            ImmutableArray<BoundExpression> rewrittenArguments = VisitArguments(
+                indexerAccess.Arguments,
+                indexer,
+                argsToParamsOpt,
+                argumentRefKinds,
+                ref transformedReceiver!,
+                out ArrayBuilder<LocalSymbol>? argumentTemps);
```

Comment:
nit: `argumentTemps` can't be null after `VisitArguments`. We don't need the null check below